src/services/ann-index.ts

```
@@ -40,29 +48,109 @@
+  
+  // Validate dimensions if expected dimension is provided
+  if (expectedDimension !== undefined && result.length !== expectedDimension) {
+    throw new Error(
+      `Vector dimension mismatch: expected ${expectedDimension}, got ${result.length}`
+    );
   }
-  return new Float32Array(vector);
+  
+  // Validate that vector contains no NaN or infinite values
+  for (let i = 0; i < result.length; i++) {
+    if (!isFinite(result[i])) {
```

Biome (1.9.4)

[error] isFinite is unsafe. It attempts a type coercion. Use Number.isFinite instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isFinite instead.


(lint/suspicious/noGlobalIsFinite)